### ISSUE TITLE
Rubik's trading inventory grows on dialogue prompt

### ISSUE THREAD

**Issue by waxcatape** [CleverRaven/Cataclysm-DDA#54225]
### Describe the bug

 A while after first establishing trade if you try talking to Rubik, his inventory gets rerolled but the old items stay. It seems to impact CBMs more than other things. The result is that you can turn into Inspector Gadget overnight on day 25 of the run provided that you have enough stuff to trade for the bionics.

### Steps to reproduce

1. Go to Rubik, trade.
2. Maybe you need access to CBMs? I'm not sure.
3. Wait enough time for the inventory to reroll (~ 2 weeks).
4. Come back and talk to Rubik about trading or installing CBMs.
5. Exit the dialogue screen and talk to Rubik again.
6. Witness the power of the Exodii, as new CBMs and other wares like thread, pipe fittings and welding tanks materialize in Rubik's inventory out of thin air.

### Expected behavior

I expected Rubik to reroll wares only after some reasonable period of time. More importantly, old wares, especially CBMs, probably shouldn't persist. 

### Versions and configuration

- OS: NixOS 21.11
- Game version: 1793043
- Graphics version: Tiles
- Ingame language: English
- Mods loaded: dda, no_npc_food, no_fungal_growth, speedydex

### Additional context

More often than not, when you try talking to Rubik, you get the following error: 

DEBUG    : loading non-main map at (-38,272,1) which overlaps with main map (abs_sub = (-42,268,1)) is not supported

 FUNCTION : void map::load(const tripoint_abs_sm&, bool, bool)
 FILE     : src/map.cpp
 LINE     : 6672
 VERSION  : 1793043

I'm not sure if the error and the unexpected inventory are related.

**Comment by Maleclypse** [comment]
So what actually is happening is that there is a trust function with Rubik as he trades with you and as that increases his inventory increases.

**Comment by waxcatape** [comment]
Closing in favour of #54407
